torch/_dynamo/variables/torch.py

```diff
@@ -721,6 +642,150 @@ def check_any_unspec(x):
+        """inline behavior of torch.nn.modules.utils._ntuple"""
+        if self.value is torch.nn.modules.utils._ntuple:
+            count = args[0].as_python_constant()
+        else:
+            count = self.value.__closure__[0].cell_contents
+        assert isinstance(count, int)
+        assert not kwargs
+
+        def handle_ntuple(value):
+            if value.has_unpack_var_sequence(tx):
+                return variables.TupleVariable(
+                    list(value.unpack_var_sequence(tx)),
+                )
+            elif value.is_python_constant():
+                # constant prop through it
+                return variables.ConstantVariable.create(
+                    torch.nn.modules.utils._ntuple(count)(value.as_python_constant()),
+                )
+            else:
+                unimplemented(f"torch.nn.modules.utils._ntuple({value})")
+
+        if self.value is torch.nn.modules.utils._ntuple:
+            return variables.LambdaVariable(handle_ntuple)
+        else:
+            return handle_ntuple(args[0])
+
+
+class TorchVariable(BaseTorchVariable):
+    """Points to a module, classes or functions in torch.*"""
+
+    def __init__(self, value, **kwargs):
+        if (
+            isinstance(value, collections.abc.Hashable)
+            and value in tensor_dunder_fns_remap
+        ):
+            value = tensor_dunder_fns_remap[value]
+
+        super().__init__(value, **kwargs)
+
+        # the remainder of this is just optional debug checks
+        try:
+            self_should_be_none = getattr(self.value, "__self__", None)
+        except RuntimeError as e:
+            assert "No such operator" in str(e), str(e)
+            self_should_be_none = None
+
+        # assert "_ntuple.<locals>.parse" not in str(value)
+
+        if self_should_be_none is None:
+            pass
+        elif isinstance(self_should_be_none, types.ModuleType):
+            # weird ones like torch.nn.functional.avg_pool2d have __self__
+            name = self_should_be_none.__name__
+            assert re.match(r"^(torch|math)([.]|$)", name), f"__self__ set to {name}"
+        elif isinstance(
+            self_should_be_none, type(torch._C._get_tracing_state.__self__)
+        ):
+            # some _C functions have __self__ as a null capsule
+            pass
+        elif isinstance(self_should_be_none, torch_special_class_types):
+            pass
+        else:
+            raise AssertionError(f"{value} found with __self__ set")
+
+    def __repr__(self):
+        return f"TorchVariable({self.value})"
+
+    def call_hasattr(self, tx, name):
+        result = hasattr(self.value, name)
+        return variables.ConstantVariable.create(result)
+
+    def python_type(self):
+        if isinstance(self.value, (torch.Tensor, torch.nn.Module, torch.device)):
+            return type(self.value)
+        if isinstance(self.value, type):
+            return type
+        return super().python_type()
+
+    def call_function(
```

Comment:
This is entirely new, it looks like?

Reply:
No, this is the original ```TorchVariable```'s ```call_function```.

Reply:
But it's not a code motion, you now have two `call_function` implementations

Reply:
Yes, because we have 4 classes in torch.py:
* BaseTorchVariable
* TorchCtxManagerClassVariable
* TorchInGraphFunctionVariable
* TorchVariable

You can search there are 3 ```def call_function``` in torch.py for each of the children class.